tests/test_annotated.py

```diff
@@ -505,28 +505,6 @@ class AnnotatedFieldModel(BaseModel):
         }
     ]
 
-    # Ensure that the inner annotation does not override the outer, even for metadata:
```

Comment:
Due to one of the bugs being fixed in this PR (related to the usage of forward references), the constraint in `Annotated` was dropped. Now the test is failing because the unexpected order (described in https://github.com/pydantic/pydantic/issues/10507) applies.